### PR TITLE
deps: update to bpmnlint-plugin-camunda-compat@2.21.1

### PR DESCRIPTION
**Pull request by marstamm**
### Proposed Changes
Integrate latest `bpmnlint-plugin-camunda-compat` release to fix loop detection for big diagrams:
 
Recording 2024-07-01 at 16 26 57

closes https://github.com/camunda/camunda-modeler/issues/4401

Try out with Diagram linked in support case: [SUPPORT-22488](https://jira.camunda.com/browse/SUPPORT-22488)
### Checklist

To ensure you provided everything we need to look at your PR:

* [x] **Brief textual description** of the changes present
* [x] **Visual demo** attached
* [x] **Steps to try out** present, i.e. [using the `@bpmn-io/sr` tool](https://github.com/bpmn-io/sr)
* [x] Related issue linked via `Closes {LINK_TO_ISSUE}` or `Related to {LINK_TO_ISSUE}`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,7 @@ ___Note:__ Yet to be released changes appear here._
 ### BPMN
 
 * `FEAT`: support converging inclusive gateway ([#3613](https://github.com/camunda/camunda-modeler/issues/3613))
+* `FIX`: improve `no-loop` performance ([bpmnlint-plugin-camunda-compat#165](https://github.com/camunda/bpmnlint-plugin-camunda-compat/pull/165))
```
Comment (nikku) [inline CHANGELOG.md:16]:
```suggestion
* `FIX`: improve `no-loop` performance ([camunda/bpmnlint-plugin-camunda-compat#165](https://github.com/camunda/bpmnlint-plugin-camunda-compat/pull/165))
```

Comment (nikku) [reply CHANGELOG.md:16]: We prefix the org usually, cf. existing changelog entries.

Comment (marstamm) [reply CHANGELOG.md:16]: I see! I thought we only prefix the org if it is different than the current one (same as github does with auto-shortening links)

**Review by jarekdanielak — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (nikku) [reply CHANGELOG.md]: Let's discuss, but be pragmatic. 

For me I'd like to be able to just copy things over from one project to another, without thinking too much (i.e. if I shall adopt the org prefix).

Comment (marstamm) [reply CHANGELOG.md]: I'm fine with this as well, was just not aware 😄

**Pull request merged**